src/Microsoft.Azure.WebJobs.Host/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs

```diff
@@ -18,14 +18,65 @@
 using Microsoft.Azure.WebJobs.Host.Triggers;
 using Microsoft.Extensions.Logging;
 using Microsoft.WindowsAzure.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Config;
+using System.Threading;
 
 namespace Microsoft.Azure.WebJobs.Host.Blobs.Triggers
 {
+    internal class BlobTriggerExtension : IExtensionConfigProvider,
+        IAsyncConverter<IStorageBlob, Stream>
+    {
+        private IStorageAccountProvider _accountProvider;
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            _accountProvider = context.Config.GetService<IStorageAccountProvider>();
+
+            var rule = context.AddBindingRule<BlobTriggerAttribute>();
+            rule.BindToTrigger<IStorageBlob>();
+
+            rule.AddConverter<IStorageBlob, DirectInvokeString>(blob => DirectInvokeString.New(blob.GetBlobPath()));
+            rule.AddConverter<DirectInvokeString, IStorageBlob>(ConvertX);
+
+            // Common converters shared between [Blob] and [BlobTrigger]
+
+            // Trigger already has the IStorageBlob. Whereas BindToInput defines: Attr-->Stream. 
+            //  Converter manager already has Stream-->Byte[],String,TextReader
+            context.AddConverter<IStorageBlob, Stream>(this);
+
+            // Blob type is a property of an existing blob. 
+            context.AddConverter(new StorageBlobToCloudBlobConverter());
+            context.AddConverter(new StorageBlobToCloudBlockBlobConverter());
+            context.AddConverter(new StorageBlobToCloudPageBlobConverter());
+            context.AddConverter(new StorageBlobToCloudAppendBlobConverter());
+        }
+
+        async Task<Stream> IAsyncConverter<IStorageBlob, Stream>.ConvertAsync(IStorageBlob input, CancellationToken cancellationToken)
+        {
+            WatchableReadStream watchableStream = await ReadBlobArgumentBinding.TryBindStreamAsync(input, cancellationToken);
```

Comment:
So is all the parameter watcher stuff still wired up for these bindings? The stuff that shows in progress blob download/upload percentage progress in the Dashboard?